app/services/github-state.js

```diff
@@ -0,0 +1,73 @@
+import Ember from 'ember';
+
+const { get, Service, inject: { service } } = Ember;
+
+function dec2hex(dec) {
+  return (`0${dec.toString(16)}`.substr(-2));
+}
+
+function generateRandomString(len) {
+  let arr = new window.Uint8Array((len || 40) / 2);
```

Comment:
An example of why the above is potentially confusing from JS itself! The `8` here is what I mean. Similar to `2` above.